gcloud/pubsub/message.py

```diff
@@ -44,6 +49,20 @@ def attributes(self):
             self._attributes = {}
         return self._attributes
 
+    @property
+    def timestamp(self):
+        """Return timestamp from attributes, if passed.
+
+        :rtype: datetime
+        :returns: timestamp (in UTC timezone) parsed from RFC 3369 timestamp
+        :raises: ValueError if timestamp not in ``attributes``, or malformed
+        """
+        stamp = self.attributes.get('timestamp')
+        if stamp is None:
+            raise ValueError('No timestamp')
```

Comment:
Callers of `Subscription.pull()`, or handlers for push-delivered messages, would do the sorting.   E.g., see the regression test changes below.

Noted in 04ced70.